pkg/ephemeral-disk-utils/utils.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package ephemeraldiskutils
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"strconv"
+
+	"kubevirt.io/kubevirt/pkg/logging"
+)
+
+func RemoveFile(path string) error {
+	err := os.Remove(path)
+	if err != nil && os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		logging.DefaultLogger().V(2).Error().Reason(err).Msg(fmt.Sprintf("failed to remove cloud-init temporary data file %s", path))
+		return err
+	}
+	return nil
+}
+func FileExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	exists := false
+
+	if err == nil {
+		exists = true
+	} else if os.IsNotExist(err) {
+		err = nil
+	}
+	return exists, err
+}
+func Md5CheckSum(filePath string) ([]byte, error) {
+	var result []byte
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		return result, err
+	}
+	defer file.Close()
+
+	hash := md5.New()
+	_, err = io.Copy(hash, file)
+
+	if err != nil {
+		return result, err
+	}
+
+	result = hash.Sum(result)
```

Comment:
Wait a sec. when did `result` get initialized? I'm pretty sure the intent was to pass in nil here.

Reply:
hash is an object with state.  The Sum() function takes in a byte slice that it appends the result to. 

this similar to something like.
```
mySlice = append(mySlice, otherSlice...)
```

Reply:
Understood. I think the thing that's confusing is you copied in the file earlier then on this line passed in a blank array.

After looking at it closer, this code probably works as intended, but I kind of feel like it's an accidental side effect of result already being empty.

Reply:
alrighty, fixed in an update i'm working on.